### PR TITLE
Binance/FTX: spot websocket fixes

### PR DESCRIPTION
**Pull request by ydm**
# PR Description
With c38757a68990858aa989557d2cdfb8be1e7f932f a number of issues were introduced. Some of them are:
1. instead of reading the order status, which is pushed through the websocket, floating point numbers were compared for equality :)
2. there was no average price populated
3. incorrectly fee was assigned to cost
4. "m" was being overridden by "M" 

This PR fixes all of them and, on top of that:
1. orders fields of WsOrderUpdateData the way they are described in the latest version of Binance docs + that's the same order when they come out of the websocket
2. adds and populates order.Detail.FeeAsset
3. sorts out mixed := and var declarations
4. orders temporary variables by the order they are used

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ydm** [comment]
@yangrq1018 , tagging you since you might be interested in this PR.

**Review by shazbert — APPROVED** [review]
Tested execution front end, approved for Binance and FTX changes.

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thank you for the update and fix! Just one minor thing

File: exchanges/binance/binance_websocket.go [hunk 1]
```diff
@@ -207,66 +207,66 @@ func (b *Binance) wsHandleData(respRaw []byte) error {
...
+				if data.Data.CumulativeFilledQuantity != 0 {
+					averagePrice = data.Data.CumulativeQuoteTransactedQuantity / data.Data.CumulativeFilledQuantity
+				}
+				remainingAmount := data.Data.Quantity - data.Data.CumulativeFilledQuantity
+				cost := averagePrice * data.Data.Quantity
+				pair, assetType, err := b.GetRequestFormattedPairAndAssetType(data.Data.Symbol)
+				if err != nil {
+					return err
+				}
+				feeAsset := currency.NewCode(data.Data.CommissionAsset)
```
[the remark this answers is not in the document]
Comment (ydm) [reply exchanges/binance/binance_websocket.go]: You're right. Just to make sure I added a check on the wsOrderUpdate handling side.

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for making those changes!

**Review by thrasher- — APPROVED** [review]
tACK, great work @ydm 👍

**Pull request merged**